tests/test_save_fields.py

```diff
@@ -93,3 +93,15 @@ def test_correctly_handle_foreignkeys_id_field_in_update_fields():
 
     tmwfk.save(update_fields=['fkey'])
     assert tmwfk.get_dirty_fields(check_relationship=True) == {}
+
+
+@pytest.mark.django_db
+def test_save_deferred_field_with_update_fields():
+    TestModel.objects.create()
+
+    tm = TestModel.objects.defer('boolean').first()
+    tm.boolean = False
+
+    assert tm.get_dirty_fields() == {}
+
+    tm.save(update_fields=['boolean'])
```

Comment:
The test you're proposing checks that a deferred field will never be considered, which is wrong.
The Django 1.10 release deeply changed the deferred fields implementation, an since then, if you explicitly set the value of a deferred field, it is not considered deferred anymore.
To resume:
Django < 1.10:
```
    TestModel.objects.create()
    tm = TestModel.objects.defer('boolean').first()
    tm.save(update_fields=['boolean'])
    tm.boolean = False
    assert tm.get_dirty_fields() == {}
```
Django >= 1.10
```
    TestModel.objects.create()
    tm = TestModel.objects.defer('boolean').first()
    tm.save(update_fields=['boolean'])
    tm.boolean = False
    assert tm.get_dirty_fields() == {'boolean': True}
```